### PR TITLE
Initial Sprite Kit support

### PR DESCRIPTION
**Pull request by LearnCocos2D**
this one ought to be properly merged with develop
216 files changed on creation

### CODE REVIEW

File: SpriteBuilder/ccBuilder/PlugInNodeCollectionView.h [hunk 1]
```diff
@@ -7,6 +7,7 @@
 //
 
 #import <Cocoa/Cocoa.h>
+#import "ProjectSettings.h"
```
Comment (vlidholt) [inline SpriteBuilder/ccBuilder/PlugInNodeCollectionView.h:10]: Imports should almost never go in the header files. Instead, place the import in the .m file. If you need to reference a class, use the @class directive. Exceptions for 'allowed' imports can be external frameworks, inheritance or for protocols.

(There are multiple places in the code, only commenting on this one.)

Comment (LearnCocos2D) [inline SpriteBuilder/ccBuilder/PlugInNodeCollectionView.h:10]: In this case it was to import an enum.

I could make separate headers for public enums though. I usually do that whenever the enum is the only reason to import a header, and the enums is used by several (many) classes.

File: SpriteBuilder/SpriteBuilder.xcodeproj/xcshareddata/xcschemes/SpriteBuilder (Debug).xcscheme [hunk 2]
```diff
@@ -58,6 +58,26 @@
          </BuildableReference>
       </BuildableProductRunnable>
       <AdditionalOptions>
+         <AdditionalOption
```
Comment (vlidholt) [inline SpriteBuilder/SpriteBuilder.xcodeproj/xcshareddata/xcschemes/SpriteBuilder (Debug).xcscheme:61]: Changes to this file shouldn't be committed. (It's the user's preferences, and default options were good)

File: SpriteBuilder/ccBuilder/PlugInManager.m [hunk 3]
```diff
@@ -173,7 +173,7 @@ - (CCNode*) createDefaultNodeOfType:(NSString*)name
 	}
     
     CCNode* node = [[editorClass alloc] init];
-    [node setUserObject: [NodeInfo nodeInfoWithPlugIn:plugin]];
+    [node setUserObject:[NodeInfo nodeInfoWithPlugIn:plugin]];
```
Comment (vlidholt) [inline SpriteBuilder/ccBuilder/PlugInManager.m:176]: Watch out for 'non-changes' that only affects formatting. Generally these shouldn't be committed.

**Comment by vlidholt** [comment]
I've added a couple of comments inline, so have a look at those too.

Will need a couple of changes before I can merge this:
- Why is SpriteBuilder/libs/SpriteKit/CCBReader added (It’s a 2 MB binary file!)? If this is a built library, the building of it should go in the build script and we only keep the source in the repo. Otherwise we would need to add a new binary for every little change, making the repository grow very large very quickly.
- Avoid updating the Cocos2d submodule (especially if it downgrades it to an older version).
- When making changes to the project files, please make a comment about what changes have been done. E.g. what has been changed in the PROJECTNAME file, and for what reason?
- I think we should move the SpriteKit CCBReader to a sub module probably.

Before making a new pull request it would be good to make a git rebase, to avoid the overhead of adding the binary file.

File: SpriteBuilder/ccBuilder/PlugInNodeCollectionView.h (hunk 1 above)
Comment (vlidholt) [inline SpriteBuilder/ccBuilder/PlugInNodeCollectionView.h:10]: You got a good point there, that can be a valid case too! :) But the enum is still not needed in the .h file in this case, or am I missing something?

Comment (LearnCocos2D) [inline SpriteBuilder/ccBuilder/PlugInNodeCollectionView.h:10]: Hmmm there should be a new property using the CCBTargetEngine enum, or a method with that enum as parameter.

File: SpriteBuilder/SpriteBuilder.xcodeproj/xcshareddata/xcschemes/SpriteBuilder (Debug).xcscheme (hunk 2 above)
Comment (LearnCocos2D) [inline SpriteBuilder/SpriteBuilder.xcodeproj/xcshareddata/xcschemes/SpriteBuilder (Debug).xcscheme:61]: should we add .xcscheme to the .gitignore list?

**Comment by LearnCocos2D** [comment]
CCBReader library/binary: no idea how it got there. I didn't move or add it there on purpose.

CCBReader submodule: agree, this should probably be a redesigned kobold kit because there'll be more things needed besides the reader and compatibility classes, like rebuilding the cocos2d-ui classes on the SK side of things, how nodes schedule updates, timeline animation manager or extensions, and so on.

I'll resend a pull request after removing the binary and rebase. I think we should do a couple smaller pull requests over the coming weeks just so I get the hang of it and can adapt my own work style to the process.

**Comment by LearnCocos2D** [comment]
created a new pull request after rebase and remove of binary: https://github.com/apportable/SpriteBuilder/pull/247

**Pull request merged**